django/db/models/functions/text.py

```diff
@@ -202,6 +202,48 @@ def as_sqlite(self, compiler, connection, **extra_context):
         return super().as_sql(compiler, connection, function='UNICODE', **extra_context)
 
 
+class RegexpReplace(Func):
+    function = 'REGEXP_REPLACE'
+
+    def __init__(self, expression, pattern, replacement=Value(''), flags=Value(''), **extra):
+        if not hasattr(pattern, 'resolve_expression'):
+            if not isinstance(pattern, (str, type(None))):
+                raise TypeError("'pattern' must be a string or None")
+            pattern = Value(pattern)
```

Comment:
FWIW the fallback of `str` to `Value(str)` might be counter intuitive since all other `Func` consider `str` as `F(str)` references.

I think we should stick to that here as well which means `Value('R\.')` should be provided instead. It might involve more boilerplate but it at least the API is consistent.

Reply:
Yup. I inherited this from the ticket, but agree that it doesn't make sense.